AmplifyPlugins/Analytics/Sources/AWSPinpointAnalyticsPlugin/AWSPinpointAnalyticsPlugin+Options.swift

```diff
@@ -0,0 +1,48 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+
+extension AWSPinpointAnalyticsPlugin {
+    public struct Options {
+        static let defaultAutoFlushEventsInterval: TimeInterval = 60
+        static let defaultTrackAppSession = true
+        static let defaultAutoSessionTrackingInterval: TimeInterval = {
+        #if os(macOS)
+            .infinity
+        #else
+            5
+        #endif
+        }()
+
+        public let autoFlushEventsInterval: TimeInterval
+        public let trackAppSessions: Bool
+        public let autoSessionTrackingInterval: TimeInterval
+
+        #if os(macOS)
+        public init(autoFlushEventsInterval: TimeInterval = 60,
+                    trackAppSessions: Bool = true,
+                    autoSessionTrackingInterval: TimeInterval = .infinity) {
```

Comment:
On a second though, are we sure we want to expose this flag, and if so with this name? 

It's usage it very unintuitive. When the app goes to the background, we set the current session to "paused" and this value is meant to be how much time it has to pass until it effectively becomes "stopped" if the app is still in the background (i.e. if the apps comes back to the foreground before this time interval, then we just "resume" the session instead of starting a new one; if more time has passed we start a new session).

But it's also not completely honoured, because we also keep track on how much the OS lets the app run in the background.
So if a customer sets this to 60 seconds, we will still stop the session once the OS stops background execution (which is around 5 seconds IIRC).

Also this not used on macOS at all. It's set to `.max` (or  `.infinity`) because we originally figured there was no limit on how long an app can be "backgrounded" in a desktop system, but eventually we decided to not pause sessions in macOS as the app still runs.

tl;dr: I don't think this value should be publicly exposed.

Reply:
I've just been focused on parity in translating the capabilities in old to the new DX (config + options) so I didn't consider whether this option should even exist, thanks for the detailed explanation!

I'm inclined to remove this. we can always add it in later with better documentation on the usage, and clearer naming. I doubt anyone is actually overriding this in the old configuration file due to how unintuitive it is. If they set this to 1 second then a session is "stopped" almost immediately after the app goes to background. I think the use case is to "create a new session immediately if the app is moved to the background".  When setting this to a large value, the OS stops background execution without any guarantees, as you've said, so the value is not completely honoured.

Reply:
Removing `autoSessionTrackingInterval` in this PR, commit https://github.com/aws-amplify/amplify-swift/pull/3567/commits/9cbc295b6a8f94c33a989c09cf9910460a0da9c2